.github/workflows/lint.yml

```diff
@@ -15,19 +15,17 @@ jobs:
       - name: Check out Git repository
         uses: actions/checkout@v2
 
-      - name: Set up node
-        uses: actions/setup-node@v1
-        with:
-          node-version: 16.x
-          registry-url: https://registry.npmjs.org
+      - name: Set the correct Node version using nvm
+        shell: bash -l {0} 
+        run: nvm install
 
       - name: Install dependencies
         run: npm ci
 
       - name: Run linters
-        uses: wearerequired/lint-action@v1
+        uses: wearerequired/lint-action@v2
         with:
           github_token: ${{ secrets.github_token }}
           prettier: true
-          auto_fix: true
-          prettier_extensions: 'css,html,js,json,jsx,md,sass,scss,ts,tsx,vue,yaml,yml,sol'
+          auto_fix: false
+          prettier_extensions: 'css,html,js,json,jsx,md,sass,scss,ts,tsx,vue,yaml,yml,sol'
```

Comment:
I strongly recommend you replace lint-action with your own `yarn lint`, as this has better versioning control, will better respect any lint configuration you set up, and will be more testable locally.

Reply:
For sure, I also feel concerned to see the non-reproducibility on local.

I actually saw smart-order-route uses npm [lint](https://github.com/Uniswap/smart-order-router/blob/main/.github/workflows/lint.yml), and curious to see your thoughts. Do you think we should strive to use `yarn` or `npm` with `nvm`?

Reply:
the web team uses `yarn`, not `npm`. we also use `nvm`, but not in our github actions workflow files.